src/Extractor/TikaTextExtractor.php

```diff
@@ -80,6 +80,10 @@ public function getContent($file)
         $path = $file instanceof File ? $this->getPathFromFile($file) : $file;
         $command = sprintf('tika %s %s', $mode, escapeshellarg($path));
         $code = $this->runShell($command, $output);
+        //Cleanup temp file
+        if ($path instanceof File) {
+            unlink($path);
```

Comment:
Wouldn’t it be a string here? That’d make this statement never execute

Reply:
You're right, that this is not the right check. It should check if `$file` is an instance of `File` instead of $path.
It actually should not unlink the `$path` because the file is only copied inside the `getPathFromFile` function.